### PR TITLE
Remove special handling to set alpha of framebuffer to 1.0

### PR DESCRIPTION
**Pull request by hkr**
### Purpose of this PR

Don't force alpha to 1.0 on Android and iOS.
This is already handled at a lower level (on Android the native window is set to opaque when not using "preserve alpha").

This avoids and extra blit in some cases and also reduces complexity in URP.

Still untested on anything XR

---
### Testing status

Manual testing with project that writes alpha to the framebuffer.

**Automated Tests**: no
**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline

---
### Overall Product Risks
**Technical Risk**: Medium?

**Halo Effect**: Medium?

### CODE REVIEW

**Comment by daves-unity** [comment]
Florian shared his project with me and I'm doing some testing on XR.  Will report back soon.

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderPipelineCore.cs [hunk 1]
```diff
@@ -24,7 +24,6 @@ namespace UnityEngine.Rendering.Universal
         public PostProcessingData postProcessingData;
         public bool supportsDynamicBatching;
         public PerObjectData perObjectData;
-        public bool killAlphaInFinalBlit;
```
Comment (alexey-unity) [inline com.unity.render-pipelines.universal/Runtime/UniversalRenderPipelineCore.cs]: this changes public api (i guess?) i dont know how we handle this in SRP

Comment (hkr) [reply com.unity.render-pipelines.universal/Runtime/UniversalRenderPipelineCore.cs]: @phi-lira any idea?
And is the variant of the blit shader that sets alpha to 1.0 also considered public API?

Comment (phi-lira) [reply com.unity.render-pipelines.universal/Runtime/UniversalRenderPipelineCore.cs]: We should mark it as obsolete and always initialize/return false.

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs [hunk 2]
```diff
@@ -345,7 +338,7 @@ static void InitializeCameraData(UniversalRenderPipelineAsset settings, Camera c
             cameraData.defaultOpaqueSortFlags = canSkipFrontToBackSorting ? noFrontToBackOpaqueFlags : commonOpaqueFlags;
             cameraData.captureActions = CameraCaptureBridge.GetCaptureActions(camera);
 
-			bool needsAlphaChannel = camera.targetTexture == null && Graphics.preserveFramebufferAlpha && PlatformNeedsToKillAlpha();
+			bool needsAlphaChannel = camera.targetTexture == null && Graphics.preserveFramebufferAlpha;
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs]: I think the Graphics.preserveFramebufferAlpha should also be removed in this case. @aleks01010101 do you know?

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs [hunk 3]
```diff
@@ -252,13 +252,6 @@ static void SetSupportedRenderingFeatures()
 #endif
         }
 
-		static bool PlatformNeedsToKillAlpha()
-		{
-			return Application.platform == RuntimePlatform.IPhonePlayer ||
-                Application.platform == RuntimePlatform.Android ||
-                Application.platform == RuntimePlatform.tvOS;
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs]: The PR mentions this is handle in low level for Android and iOS. What about tvOS?

**Comment by phi-lira** [comment]
Please also add changelog entry.

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs (hunk 2 above)
Comment (aleks01010101) [reply com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs]: No, I don't think so. We can use RG11B10 when this is false.

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs (hunk 3 above)
Comment (hkr) [reply com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs]: I will ask QA to check, but I assume it's same as iOS. @alexey-unity?

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderPipelineCore.cs (hunk 1 above)
Comment (hkr) [reply com.unity.render-pipelines.universal/Runtime/UniversalRenderPipelineCore.cs]: 👍

**Comment by hkr** [comment]
> This is great. Thanks for working on this. Is this supposed to work on all Android and iOS versions we support?

Yes, should work on all devices we support. Only thing that could hold this up could be QA pass on old weird devices that I don't have. But since Android 4.4 is the oldest version we support now I'm confident that this should work.

**Comment by hkr** [comment]
> The public API cannot be removed. That's the only blocking change I'm requesting. Added more comments.
Done

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs (hunk 3 above)
Comment (alexey-unity) [reply com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs]: tvos = ios in this regard

**Comment by sharlenet** [comment]
Even though I only made a minor change to CHANGELOG.md, GitHub somehow thinks that I've changed the entire file. I used the web interface from a windows machine, and all the line breaks were changed.

**Pull request merged**